### PR TITLE
Places Dynamo persistence

### PR DESCRIPTION
**Pull request by jancona**
Things to consider:
* Do we have a use case for `SelectPlacesByID`? I don't see anything using it, and it's hard to implement with the key structure I was going to use to implement `SelectPlacesByFullNamePrefix`. I can look at other schema changes to make it work, but if we don't really need it, I can either remove it or implement it inefficiently, i.e. by calling `SelectPlace` in a loop.
* This PR changes the partition key on the DynamoDB table to be a string instead of an integer. To deploy this, we will need to drop the DynamoDB table and re-create it. This will lose data unless I write a migration. AFAIK, ourroots.anconafamily.com is the only deploy using Dynamo, so I don't think we need to migrate. 
* In reviewing the schema, I noticed that many of our entities only have a single partition for the GSI. That's probably okay for small entities like Category, but I need to think about things like Record where the GSI PK is just "record". I can think of several potential fixes for this, but we should probably decide on one soon, because doing anything would require loss of data or a migration.

### CODE REVIEW

**Comment by jancona** [comment]
I made this a draft PR because I haven't deployed it to AWS yet, and I may want to wait until we're sure of the schema changes.

**Comment by DallanQ** [comment]
SelectPlacesByID is used in stdplace/standardizer.go. It makes two persister calls: SelectPlaceWordsByWord, and SelectPlacesByID. I'm fine if the persister code for dynamodb implements that using a loop that looks up places one at a time.

Agreed - we don't need to migrate

Let's talk about changing the GSI PKs - how about right after our meeting on Tuesday? We could also talk on Monday if you prefer.

**Comment by jancona** [comment]
Got it. I missed that usage before--perhaps I hadn't merged the commit. Some of the solutions to the GSI partition issue might take care of the multi-select issue too. I'll try to put together some notes before we discuss it. Tuesday is better than Monday for me, so let's go with after the meeting.

**Comment by DallanQ** [comment]
Sounds good - thanks!

File: model/place.go [hunk 1]
```diff
@@ -23,13 +23,15 @@ type Uint32Slice []uint32
...
+	Type             string      `json:"-" dynamodbav:"sk"`
+	AltSort          string      `json:"-" dynamodbav:"altSort"`
 	Name             string      `json:"name"`
-	FullName         string      `json:"fullName"`
-	AltNames         StringSlice `json:"altNames"`
-	Types            StringSlice `json:"types"`
+	FullName         string      `json:"fullName" dynamodbav:"fullName"`
+	AltNames         StringSlice `json:"altNames" dynamodbav:"altNames"`
+	Types            StringSlice `json:"types" dynamodbav:"types"`
 	LocatedInID      uint32      `json:"locatedInId"`
```
Comment (DallanQ) [inline model/place.go:33]: just curious, why not have a dynamodbav annotation on LocatedInID, or CountryID?

File: model/place.go [hunk 2]
```diff
@@ -23,13 +23,15 @@ type Uint32Slice []uint32
 
 // Place holds information about a place
 type Place struct {
-	ID               uint32      `json:"id" dynamodbav:"pk"`
+	ID               uint32      `json:"id" dynamodbav:"pk,string"`
+	Type             string      `json:"-" dynamodbav:"sk"`
```
Comment (DallanQ) [inline model/place.go:27]: Just for my understanding, these fields: Place.Type + AltSort, PlaceWord.Pk + Type, and PlaceSettings.Pk + Sk are only used if the backend is dynamodb?

**Comment by DallanQ** [comment]
LGTM. Just had a couple of questions. Feel free to go ahead and merge.

File: model/place.go (hunk 1 above)
Comment (jancona) [reply model/place.go:33]: It defaults to using the names in the `json` annotation, so the better question is why FullName, AltNames and Types have dynamodbav annotations. I'll look at that.

File: model/place.go (hunk 2 above)
Comment (jancona) [reply model/place.go:27]: Yes, the ones with `dynamodbav ` and `json:"-"` annotations are DynamoDB-specific.

**Pull request merged**